omnigibson/envs/env_base.py

```diff
@@ -224,10 +227,19 @@ def _load_robots(self):
                 # Import the robot into the simulator
                 og.sim.import_object(robot)
 
+            # Auto-initialize all robots
+            og.sim.step()
+
+            self.scene.update_initial_state()
+            og.sim.stop()
+
     def _load_objects(self):
         """
         Load any additional custom objects into the scene
         """
+        assert og.sim.is_stopped()
+        og.sim.play()
```

Comment:
same thing here

Reply:
though now that i think about it, i may see why you chose to do this way -- we want the scene to be restored properly, but we can't actually load the state if we add more objects to the scene, thereby invalidating the len(state) == num_objects check...hmmm....

Reply:
i wonder if there's a clean way to resolve this

Reply:
Let's discuss in person.